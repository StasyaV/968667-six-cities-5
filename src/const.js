export const cities = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];
export const roomTypes = [`Apartment`, `Private Room`, `House`, `Hotel`];
