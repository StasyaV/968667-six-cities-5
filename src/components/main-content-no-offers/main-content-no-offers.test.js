import React from "react";
import renderer from "react-test-renderer";
import MainContentNoOffers from "./main-content-no-offers";
import {cities} from "../../const";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import history from "../../browser-history";

const mockStore = configureStore([]);

describe(`MainContentNoOffers render`, () => {
  const initialState = {
    CITIES: {
      city: `Amsterdam`,
      cities
    }
  };
  const store = mockStore(initialState);
  it(`Should MainContentNoOffers render correctly`, () => {
    const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <MainContentNoOffers
              city={`Amsterdam`}
              cities={cities}
            />
          </Router>
        </Provider>,
        {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
