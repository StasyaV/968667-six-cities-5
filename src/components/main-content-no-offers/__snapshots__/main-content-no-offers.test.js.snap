// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContentNoOffers render Should MainContentNoOffers render correctly 1`] = `
<main
  className="page__main page__main--index page__main--index-empty"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
            onClick={[Function]}
          >
            <span>
              Paris
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
            onClick={[Function]}
          >
            <span>
              Cologne
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
            onClick={[Function]}
          >
            <span>
              Brussels
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="#"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
            onClick={[Function]}
          >
            <span>
              Hamburg
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
            onClick={[Function]}
          >
            <span>
              Dusseldorf
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="cities__places-container cities__places-container--empty container"
    >
      <section
        className="cities__no-places"
      >
        <div
          className="cities__status-wrapper tabs__content"
        >
          <b
            className="cities__status"
          >
            No places to stay available
          </b>
          <p
            className="cities__status-description"
          >
            We could not find any property available at the moment in 
            Amsterdam
          </p>
        </div>
      </section>
      <div
        className="cities__right-section"
      />
    </div>
  </div>
</main>
`;
